deploy/lib/deployer/repo.rb

```diff
@@ -0,0 +1,138 @@
+class Deployer
+  class Repo
+    def initialize(name, config:)
+      @name = name
+      @config = config
+    end
+
+    def update_package
+      run
+      log "Successfully updated #{package_name} to #{version} in #{name} (https://github.com/#{owner}/#{name}/pull/#{pr_number})"
+    rescue StandardError => e
+      cleanup
+      raise e
+    end
+
+    attr_reader :name
+
+    private
+
+    attr_reader :config, :pr_number
+
+    def run
+      clone_repo
+      Dir.chdir(name) do
+        create_branch
+        run_upgrade_command
+        commit_and_push_changes
+        create_pr
+        automerge_pr
+      end
+      cleanup
+    end
+
+    def clone_repo
+      url = "github.com/#{owner}/#{name}.git"
+      log "Cloning #{name}"
+      stdout, status =
+        Open3.capture2(
+          "git clone https://#{config.github_token}:x-oauth-basic@#{url} --depth=1"
+        )
```

Comment:
2 questions.

1. Does `capture2` get us all the info we want? My read of the docs says that _only_ gets us `stdout`. Do we want to use `capture2e` or `capture3` and potentially have access to `stderr` too? This sort of applies to the whole file
2. Based on the definition of `#automerge_pr`, it looks like we've already got a `gh` command preconfigured, so could we use that to clone here?

```
"gh repo clone #{owner}/#{name}"
```

Reply:
I pulled out command line commands functionality to allow for consistency of error reporting and logging.  I also used capture3 per your comment.